dw-ripple.js

```diff
@@ -54,17 +94,153 @@ export class DwRipple extends Ripple {
       /**
        * Set to `true` if ripple should not be shown
        */
-      disabled: { type: Boolean, reflect: true }
+      disabled: { type: Boolean, reflect: true },
+
+      /**
+       * Set to true when ripple shows as an unbounded.
+       */
+      unbounded: { type: Boolean, reflect: true }
     }
   }
 
   constructor(){
     super();
-    this.primary=false;
-    this.secondary = false;
     this.disabled = false;
+    this.unbounded = false;
+    this.__activeRipple = this.__activeRipple.bind(this);
+    this.__inactiveRipple = this.__inactiveRipple.bind(this);
+
+    /**
+     * Ripple animation is completed or not.
+     */
+    this.__isRippleComplete;
```

Comment:
Instead of this 2 instance variables `__isRippleComplete` and `__removeRipple`, we should have just 1 instance variable `__waitForEntryAnimation`. It's data-type is Promise. Default value is the Promise which is resolved immediately. Later on, it's value will be changed when entry animation is started (`__scale`). And when entry animation is completed, that promise gets resolved.